core/client/index.ts

```diff
@@ -74,16 +84,50 @@ export class Core {
    *
    * @private
    */
-  async sendMessage(dstIdkeys: string[], payload: string) {
+  async sendMessage(dstIdkeys: string[], messagePayload: string) {
     let batch: outboundEncPayloadType[] = new Array();
   
+    let ownId = this.olmWrapper.getIdkey();
+    let consistencyLoopback = true;
+    for (let r of dstIdkeys) {
+      if (r === ownId) {
+        consistencyLoopback = false;
+        break;
+      }
+    }
+
+    if (consistencyLoopback) {
+      dstIdkeys.push(ownId);
+    }
+
+    let sortedDstIdkeys = this.messageChains.sort_recipients(dstIdkeys);
+
     console.log("sending to...");
-    console.log(dstIdkeys);
-  
-    for (let dstIdkey of dstIdkeys) {
+    console.log(sortedDstIdkeys);
+
+    for (let dstIdkey of sortedDstIdkeys) {
+      let preEncPayload = {
+        message: messagePayload,
+        // TODO: this should be moved outside of the encrypted payload. The
+        // server is exposed to this information anyways. If a client were to
+        // send an incorrect recipients list, this could invalidate the pairwise
+        // hash-chains maintained with an arbitrary number of other clients in
+        // the system.
+        recipients: sortedDstIdkeys,
+        consistencyLoopback: consistencyLoopback,
+      };
```

Comment:
I don't understand the reasoning for why these fields should be unencrypted?

Reply:
Answer from Leon: a byz client can purposefully send an incorrect recipients list to a client, which would disrupt communication between the recipient client and any clients in that faulty recipient list that didn't actually receive the message, so we want to double check this (kind of as an optimization). If the server colludes with the byz client, communication will be disrupted with those other clients